### ISSUE TITLE
Suggestion: Define a definitive 'domain' that proves the list is working.

### ISSUE THREAD

**Issue by Gerard-ODonnell** [StevenBlack/hosts#1573]
There are many items here, that may be replicated by another list. How do I tell that **this** list is working? 

Here's an example of the first few lines of the _hosts with the fakenews extension_

```
# Title: StevenBlack / hosts with the fakenews extension
#
# This hosts file is a merged collection of hosts from reputable sources,
# with a dash of crowd sourcing via GitHub
#
# Date: 11 March 2021 02:08:33 (UTC)
# Extensions added to this file: fakenews
# Number of unique domains: 69,976
#
# Fetch the latest version of this file: https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/fakenews/hosts
# Project home page: https://github.com/StevenBlack/hosts
# Project releases: https://github.com/StevenBlack/hosts/releases
#
# =================================================== ==============

127.0.0.1 localhost
127.0.0.1 localhost.localdomain
127.0.0.1 location
255.255.255.255 broadcasthost
:: 1 localhost
:: 1 ip6-localhost

```

I propose for each individual list we include an intentionally fake and unique domain that when requested proves that **this** list is working, because it would have only ever been included in **this** list:

So:

```
# Title: StevenBlack / hosts with the fakenews extension
#
# This hosts file is a merged collection of hosts from reputable sources,
# with a dash of crowd sourcing via GitHub
#
# Date: 11 March 2021 02:08:33 (UTC)
# Extensions added to this file: fakenews
# Number of unique domains: 69,976
#
# Fetch the latest version of this file: https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/fakenews/hosts
# Project home page: https://github.com/StevenBlack/hosts
# Project releases: https://github.com/StevenBlack/hosts/releases
#
# To prove that this list is working, visit: https://StevenBlack_FakeNews.ThisDomainIsntReal.com
# If the domain doesn't resolve, then the list isn't working.
# If the domain does resolve, then the list is working.
0.0.0.0 StevenBlack_FakeNews.ThisDomainIsntReal.com
# =================================================== ==============

127.0.0.1 localhost
127.0.0.1 localhost.localdomain
127.0.0.1 location
255.255.255.255 broadcasthost
:: 1 localhost
:: 1 ip6-localhost
```

So now I can determine that things are working by visiting https://StevenBlack_FakeNews.ThisDomainIsntReal.com

**Comment by funilrys** [comment]
This is a must Steve @StevenBlack :+1: 

I will prefer a self-controlled (by you) subdomain/domain but the provided solution `stevenblack_xxxxx.domain.com` is "acceptable".

**Comment by lightswitch05** [comment]
I do this in my lists. First, I have a generic one that all my lists get `block-test.developerdan.com`, and then I have a list specific domain (ads & tracking) `block-test-ads-tracking.developerdan.com`.

**Comment by Gerard-ODonnell** [comment]
> I do this in my lists. First, I have a generic one that all my lists get `block-test.developerdan.com`, and then I have a list specific domain (ads & tracking) `block-test-ads-tracking.developerdan.com`.

I just tried your domains.

`block-test.developerdan.com` works (to show you that this particular list that holds this entry is not working).
`block-test-ads-tracking.developerdan.com` redirects to github.developerdan.com so it's not working to identify that a unique list is used as this thread is suggesting.

**Comment by Gerard-ODonnell** [comment]
> This is a must Steve @StevenBlack 👍
> 
> I will prefer a self-controlled (by you) subdomain/domain but the provided solution `stevenblack_xxxxx.domain.com` is "acceptable".

I would prefer a domain controlled by Steven Black also, but am offering to set something up if not.

**Comment by lightswitch05** [comment]
Sorry, I don't want to derail this conversation and talk about my lists. Please feel free to open a ticket in my project here https://github.com/lightswitch05/hosts/issues - long story short I created the list-specific domains, but I never got around to creating a actual page for it - which brings us back to this ticket.

* The `block-test.developerdan.com` is kinda useful for people to make sure they are using my list. Although how much value it really adds is questionable. Just pick a domain from the list and try to go to it? I know that is what the PiHole support team always says and its valid. Another issue that regularly comes up is people add my list then immediately try to visit the domain - which is already cached - and so it causes some heartache with people who can't wait for caches to be cleared. OMG THE LIST ISN'T WORKING! It is nice just having the single domain that is in all lists, that removes the whole back-and-forth of 'well, which list are you trying to use?'.
* The list-specific domains have obviously not been useful since I've never set them up. This is the first time I've ever mentioned having them - although they have been there for a long time - and I already have someone saying they are broken ;) (which, to be fair, they are broken)

In the end of the day, how useful these domains are is questionable. I suppose for the people that it works well for, I never get feedback from them, so I certainly don't have a clear picture of them providing value. Its also been somewhat useful for me to see if another list is including my list. But since its a clearly labeled test domain and is listed on my main website - it would be easy for someone to remove as well. I think its somewhat important that it is a real domain that doesn't respond with NXDOMAIN, otherwise its not a great test.

TL;DR: Its easy to do these test domains and some people probably find them useful, but how useful they are in reality is an unknown and it doesn't come without a certain amount of confusion and people saying its not working when it is just a cache issue.

**Comment by Gerard-ODonnell** [comment]
> TL;DR: Its easy to do these test domains and some people probably find them useful, but how useful they are in reality is an unknown and it doesn't come without a certain amount of confusion and people saying its not working when it is just a cache issue.

It's my opinion that if you go to the level of effort to read the header comments of this list, then you probably know what you're doing to discover this test-list-domain in the first place. You seem to run your own lists though, so perhaps your practical experience of increased real administrative overhead makes this a dumb idea.

If you explicitly set up a 5 min TTL on the A record, this might alleviate some of the caching issues.

**Comment by lightswitch05** [comment]
Thanks @Gerard-ODonnell - it currently does have a 5 min TTL. But its also proxied through the free CloudFlare plan - so I do not have the ability to make changes to it. One thing of interest is that browsers have their own DNS cache and that they purposefully refuse to respect very low TTLs. This is to help protect again DNS rebinding attacks. So there is always going to be some level of issues with cache.

Back on topic with this ticket. I just want to make it clear that I don't have any skin in this game as to supporting or not supporting this request. I just wanted to share my experience with these types of domains as a fellow blocklist maintainer. I'm glad I have them in my list. The value they provide is questionable, but I also only hear about things when they are broken, so lots of bias there.

**Comment by Gerard-ODonnell** [comment]
Updated the original request to remove the real domain idea, because it leads to administrative overhead for no benefit. A fake non-existent domain is sufficient for proving the list is working, and requires only a single line in the list.

**Comment by StevenBlack** [comment]
Thank you for the suggestion Gerard @Gerard-ODonnell.

I don't understand what benefits would arise for both the consumer and the producer perspectives.

Because "the list" is versioned every few days, and I don't understand why knowing "this list" is in the protection layer without knowing the version (the age) of the list, which is very important.

You could confirm that "the list" is in the layer but if the list it a three year old version, what have you got?  Answer: not very much.  You get a false positive.

**Comment by Gerard-ODonnell** [comment]
The advantage (for me at least) is that I get to positively confirm that the list is included in my (combined with others) blocklist. I am not so concerned that the list is up-to-date, however this could also be verified by:

0.0.0.0 StevenBlack_FakeNews_2021_03_12.ThisDomainIsntReal.com

and each day the list is published of course, the script changes the year, month, day.

**Comment by dnmTX** [comment]
Steve @StevenBlack whatever you decide to do just please don't change anything with the version release in your [package.json](https://raw.githubusercontent.com/StevenBlack/hosts/master/package.json) file.
I also second the opinion here of the questionable importance of the request here.
Would also incourage @lightswitch05 to implement **version feature** in his list(if possible) as it works like charm to check for new release when fetching 👍

**Comment by Gerard-ODonnell** [comment]
> I also second the opinion here of the questionable importance of the request here.

Perhaps for your use case, but please try to understand that there are multiple uses for such lists. One example is that many people subscribe to more than one list in the same way that virustotal.com runs files through many anti-virus scanners because any one anti-virus scanner won't catch all the viruses. In my case, I want to know that the list is included by performing a web request, which is the true _functional_ test that should be performed to test any list works.

Some points:
- I'm only suggesting a single line extra, maybe two, because of how the discussion has matured. It shouldn't take so long to implement.
- I can't see how [package.json](https://raw.githubusercontent.com/StevenBlack/hosts/master/package.json) wouldn't need to be modified to accommodate the request.
- Including such a line will have zero effect on the usual function of the list.
- I think there are benefits in including these for unit tests purposes, and from this point of view, it is useful for consumer and producer unit tests.
- Including **0.0.0.0 StevenBlack_FakeNews.ThisDomainIsntReal.com** would help someone to positively identify that the list has been included in the combined list that is made (it is time independent).
- Including **0.0.0.0 StevenBlack_FakeNews_2021_03_12.ThisDomainIsntReal.com** would help someone to identify the release date of the list. For that matter, with a unit test you could then check also the situation that the day previous **StevenBlack_FakeNews_2021_03_11.ThisDomainIsntReal.com** shouldn't exist, and if it does it will give you a heads-up that some other list you subscribe to has combined an older version of this (current) list, which will be important to catch deletions from the list that otherwise would still exist.

**Comment by uglylilduck21** [comment]
#1591
